owlkettle/guidsl.nim

```diff
@@ -220,6 +250,14 @@ proc gen(node: Node, stmts, parent: NimNode) =
       body.add(newLetStmt(name, newCall(widgetTyp)))
       for child in node.children:
         child.gen(body, name)
+      
+      let hasStateRef = not node.stateRef.isNil()
+      if hasStateRef:
+        let refVar = node.stateRef
+        let refAssignment = quote do:
+          `name`.stateRef = `refVar`
+        body.add(refAssignment)
+      
```

Comment:
```suggestion
      if not node.stateRef.isNil:
        body.add: quote do:
          `name`.stateRef = `node.stateRef`
      
```

complexity left over from when you were using optionals.

Reply:
>           `name`.stateRef = `node.stateRef`
You can't do this replacement, this won't compile due to
> /home/isofruit/dev/owlkettle/owlkettle/guidsl.nim(144, 20) Error: type mismatch: got 'array[0..-1, empty]' for '[]' but expected 'seq[Node]'

I assume this is quote do being inconsistent in its behavior if you don't have an explicit var that you want to insert.